src/CommunityToolkit/Xamarin.CommunityToolkit/Views/ViewSwitcher/TextSwitcherRenderer.android.cs

```diff
@@ -0,0 +1,568 @@
+				UpdateText();
+			else if (e.PropertyName == Label.FontProperty.PropertyName)
+				UpdateText();
+			else if (e.PropertyName == Label.LineBreakModeProperty.PropertyName)
+				UpdateLineBreakMode();
+			else if (e.PropertyName == Label.CharacterSpacingProperty.PropertyName)
+				UpdateCharacterSpacing();
+			else if (e.PropertyName == Label.TextDecorationsProperty.PropertyName)
+				UpdateTextDecorations();
+			else if (e.PropertyName == Label.TextProperty.PropertyName ||
+				e.PropertyName == Label.FormattedTextProperty.PropertyName ||
+				e.PropertyName == Label.TextTransformProperty.PropertyName)
+				UpdateText();
+			else if (e.PropertyName == Label.LineHeightProperty.PropertyName)
+				UpdateLineHeight();
+			else if (e.PropertyName == Label.MaxLinesProperty.PropertyName)
+				UpdateMaxLines();
+			else if (e.PropertyName == Label.PaddingProperty.PropertyName)
+				UpdatePadding();
+			else if (e.PropertyName == ViewSwitcher.TransitionDurationProperty.PropertyName ||
+				e.PropertyName == ViewSwitcher.TransitionTypeProperty.PropertyName)
+				UpdateTransition();
+		}
+
+		void UpdateColor()
+		{
+			var color = Element.TextColor;
+			if (color == lastUpdateColor)
+				return;
+			lastUpdateColor = color;
+
+			if (color.IsDefault)
+				children.ForEach(c => c.SetTextColor(labelTextColorDefault));
+			else
+			{
+				var androidColor = color.ToAndroid();
+				children.ForEach(c => c.SetTextColor(androidColor));
+			}
+		}
+
+		void UpdateFont()
+		{
+#pragma warning disable 618 // We will need to update this when .Font goes away
+			var f = Element.Font;
+#pragma warning restore 618
+
+			var newTypeface = f.ToTypeface();
+			if (newTypeface != lastTypeface)
+			{
+				children.ForEach(c => c.Typeface = newTypeface);
+				lastTypeface = newTypeface;
+			}
+
+			var newTextSize = f.ToScaledPixel();
+			if (newTextSize != lastTextSize)
+			{
+				children.ForEach(c => c.SetTextSize(ComplexUnitType.Sp, newTextSize));
+				lastTextSize = newTextSize;
+			}
+		}
+
+		void UpdateTextDecorations()
+		{
+			if (!Element.IsSet(Label.TextDecorationsProperty))
+				return;
+
+			var textDecorations = Element.TextDecorations;
+
+			if ((textDecorations & TextDecorations.Strikethrough) == 0)
+				children.ForEach(c => c.PaintFlags &= ~PaintFlags.StrikeThruText);
+			else
+				children.ForEach(c => c.PaintFlags |= PaintFlags.StrikeThruText);
+
+			if ((textDecorations & TextDecorations.Underline) == 0)
+				children.ForEach(c => c.PaintFlags &= ~PaintFlags.UnderlineText);
+			else
+				children.ForEach(c => c.PaintFlags |= PaintFlags.UnderlineText);
+		}
+
+		void UpdateGravity()
+		{
+			Label label = Element;
+
+			var gravity = label.HorizontalTextAlignment.ToHorizontalGravityFlags() | label.VerticalTextAlignment.ToVerticalGravityFlags();
+			children.ForEach(c => c.Gravity = gravity);
+
+			lastSizeRequest = null;
+		}
+
+		void UpdateCharacterSpacing()
+		{
+			if ((int)Build.VERSION.SdkInt >= 21)
+			{
```

Comment:
The `Build.VERSION.SdkInt` does a native call, this is expensive. Since the SDK version will not change during app execution we can cache it.
As mentioned [here](https://discord.com/channels/732297728826277939/732297916680765551/855568958626725913) on Maui channel (discord)
```suggestion
		readonly int sdkVersion = (int)Build.VERSION.SdkInt;

		void UpdateCharacterSpacing()
		{
			if (sdkVersion >= 21)
			{
```

Reply:
done